src/wait-for-element-to-be-removed.js

```diff
@@ -0,0 +1,64 @@
+import {getDocument, getSetImmediate, newMutationObserver} from './helpers'
+
+function waitForElementToBeRemoved(
+  callback,
+  {
+    container = getDocument(),
+    timeout = 4500,
+    mutationObserverOptions = {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      characterData: true,
+    },
+  } = {},
+) {
+  return new Promise((resolve, reject) => {
+    if (typeof callback !== 'function') {
+      reject(
+        'waitForElementToBeRemoved requires a function as the first parameter',
+      )
+    }
+
+    // Check if the element is not present synchronously,
+    // As the name waitForElementToBeRemoved should check `present` --> `removed`
+    try {
+      const result = callback()
+      if (!result) {
+        onDone(new Error('Element is not present in the DOM.'), true)
+      }
+    } catch (error) {
+      onDone(new Error('Element is not present in the DOM.'), true)
```

Comment:
I think this should be `onDone(error)` so we don't swallow the error. Also, what's the purpose of `true` as the second argument here?

Reply:
@ken Won't it be confusing for users? by doing `onDone(error)`? Because then they would not know the error is telling them that `waitForElementToBeRemoved` requires that the element(s) exist before waiting for removal.

Reply:
The perfect solution would have been to `instanceof` check for the `GetElementError`. But we do not have our own error class right?

I will just send back the error for now. But, making our own error class and checking for that, if the error is `GetElementError` send the above message else sending back the caught error itself seems a good solution to me.